onnxruntime/python/onnxruntime_pybind_state.cc

```diff
@@ -1580,6 +1580,7 @@ void addObjectMethods(py::module& m, ExecutionProviderRegistrationFn ep_registra
       .def_static("fpga", []() { return OrtDevice::FPGA; })
       .def_static("npu", []() { return OrtDevice::NPU; })
       .def_static("dml", []() { return OrtDevice::GPU; })
+      .def_static("webgpu", []() { return OrtDevice::GPU; })
```

Comment:
Are you sure you want to use OrtDevice::GPU here? I'm already [facing issues](https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/python/onnxruntime_pybind_state.cc#L286) with this decision when trying to pack DML in the same pkg as the ORT gpu pkg.

Reply:
Does OrtDevice needs another field for the device 'user'/'owner'/<insert better name here>?

CUDA for CUDA/TRT, AMD for ROCm/MiGraphX, Ascend, QNN, WebGPU, etc.

That might also clean up the MemType struct so only need one value for 'pinned'.

Reply:
If we don't want to modify the definition of OrtMemoryInfo, we can use `OrtMemoryInfo.name`. Currently for WebGPU this is always "WebGPU_Buffer":

include\onnxruntime\core\framework\allocator.h
constexpr const char* WEBGPU_BUFFER = "WebGPU_Buffer";

Reply:
I think the idea above only works for ORT core. It seems does not work for the python binding.

Reply:
we use OrtDevice::GPU and just followed what CUDA and DML are using. 
Can change it and add a new type of course.
But I'm not sure what the real problem with GPU is.

Reply:
If we decide to package webgpu along with tensorrt, cuda and DML in the same ORT pkg, this function will always return "DML" as the device name (since ORT will be compiled with USE_DML). This has repercussions specifically for Python bindings. I'm fixing this for DML in my PR. But even after that fix, it'll return "CUDA" since ORT will be compiled with USE_CUDA.